shotover-proxy/tests/cassandra_int_tests/prepared_statements.rs

```diff
@@ -20,66 +18,42 @@ async fn delete(session: &CassandraConnection) {
     .await;
 }
 
-async fn insert(session: &CassandraConnection) {
-    let prepared = session
-        .prepare("INSERT INTO test_prepare_statements.table_1 (id, x, name) VALUES (?, ?, ?);");
+fn insert(session: &CassandraConnection) {
+    let prepared = session.prepare("INSERT INTO test_prepare_statements.table_1 (id) VALUES (?);");
 
-    let mut statement = prepared.bind();
-    statement.bind_int32(0, 1).unwrap();
```

Comment:
Our tests here are becoming worse with this change, we no longer test binding a string variable.
But the original tests are pretty lackluster in the first place so im not super worried.

That said, what is your plan for extending execute_prepared into an API that will allow more coverage? Might be worth leaving a TODO so we have some idea of the intention down the line.

If we decide its not worth abstracting the APIs here maybe we could just match on the connection and handle each type seperately?
Otherwise something like this might be the best way forward?
```rust
    pub fn execute_prepared(
        &self,
        prepared_query: &PreparedQuery,
        value: Vec<ResultValue>
    ) -> Vec<Vec<ResultValue>> {
```

Reply:
I could have it similar to how I had the batch queries previously. 

You could pass the query and the variables you wanted to bind and handle for each driver type separately. Would be annoying to handle separate types so we could just test binding ints, which should be enough testing for our purposes.

Reply:
I dont think just testing ints is enough, I think we would want to properly exercise the protocol level handling of different types here.

Gonna go ahead and resolve this, as theres nothing to really block on here, its just future work.